core/src/test/scala/unit/kafka/network/RequestConvertToJsonTest.scala

```diff
@@ -0,0 +1,199 @@
+    val serializeRecords: Boolean = false;
+    val request = ProduceRequest.Builder.forMagic(2, 0.toShort, 0, produceDataMap, "").build()
+    val actualNode = RequestConvertToJson.request(request, serializeRecords)
+
+    val requestData = new ProduceRequestData()
+    val expectedNode = ProduceRequestDataJsonConverter.write(requestData, version, serializeRecords)
+
+    assertEquals(expectedNode, actualNode)
+  }
+
+  @Test
+  def testFormatOfOffsetsForLeaderEpochResponseNode(): Unit = {
+    val endOffsetMap = new util.HashMap[TopicPartition, EpochEndOffset]
+    endOffsetMap.put(new TopicPartition("topic1", 0), new EpochEndOffset(1, 10L))
+
+    val version: Short = ApiKeys.OFFSET_FOR_LEADER_EPOCH.latestVersion
+    val response = new OffsetsForLeaderEpochResponse(endOffsetMap)
+    val actualNode = RequestConvertToJson.response(response, version, true)
+
+    val requestData = OffsetForLeaderEpochResponseDataJsonConverter.read(actualNode, version)
+    val expectedNode = OffsetForLeaderEpochResponseDataJsonConverter.write(requestData, version, true)
+
+    assertEquals(expectedNode, actualNode)
+  }
+
+  @Test
+  def testFormatOfProduceResponseNode(): Unit = {
+    val responseData = new util.HashMap[TopicPartition, ProduceResponse.PartitionResponse]
+    val partResponse = new ProduceResponse.PartitionResponse(Errors.NONE, 10000, RecordBatch.NO_TIMESTAMP, 100, Collections.singletonList(new ProduceResponse.RecordError(3, "Record error")), "Produce failed")
+    responseData.put(new TopicPartition("topic1", 0), partResponse)
+
+    val version: Short = ApiKeys.PRODUCE.latestVersion
+    val response = new ProduceResponse(responseData)
+    val actualNode = RequestConvertToJson.response(response, version, true)
+
+    val requestData = ProduceResponseDataJsonConverter.read(actualNode, version)
+    val expectedNode = ProduceResponseDataJsonConverter.write(requestData, version, true)
+
+    assertEquals(expectedNode, actualNode)
+  }
+
+  @Test def testFieldsRequestDescMetrics(): Unit = {
+    val expectedFields = Set("requestHeader", "request", "response", "connection",
+      "totalTimeMs", "requestQueueTimeMs", "localTimeMs", "remoteTimeMs", "throttleTimeMs",
+      "responseQueueTimeMs", "sendTimeMs", "securityProtocol", "principal", "listener",
+      "clientInformation", "softwareName", "softwareVersion", "temporaryMemoryBytes", "messageConversionsTime")
+
+    val req = request(new AlterIsrRequest(new AlterIsrRequestData(), 0))
+    val byteBuffer = req.body[AbstractRequest].serialize(req.header)
+    val send = new NetworkSend(req.context.connectionId, byteBuffer)
+    val headerLog = RequestConvertToJson.requestHeaderNode(req.header)
+    val res = new RequestChannel.SendResponse(req, send, Some(headerLog), None)
+
+    val node = RequestConvertToJson.requestDescMetrics(req.header, res, req.loggableRequest, req.context, req.session,
+      1, 1, 1, 1, 1, 1, 1, 1, 1).asInstanceOf[ObjectNode]
+    val foundFields = getFieldNames(node)
```

Comment:
I've modified the test to validate the values, but I'm not sure the worth in doing this especially when constructing the expected output is just copying the code for the node construction into the test, unless you were thinking of another way of constructing the node. Plus it's kind of a hassle when we try to modify `requestDescMetric` because it would need to be changed in 2 places: the code and the test.